File: src/leetcode/solved-collection/704. Binary Search/binary_search_704.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1

        while start <= end:
            mid = int((end + start) / 2)
            print(mid)

            if nums[mid] == target:
                return mid

            if target > nums[mid]:
                start = mid + 1
            elif target < nums[mid]:
                end = mid - 1

        return -1

File: src/leetcode/solved-collection/704. Binary Search/binary_search_704.go
func search(nums []int, target int) int {
    start := 0
    end := len(nums) - 1

    for start <= end {
        mid := (start + end) / 2

        if nums[mid] == target {
            return mid
        } else if target > nums[mid] {
            start = mid + 1
        } else if target < nums[mid] {
            end = mid - 1;
        }
    }

    return -1;

}
